### ISSUE TITLE
Query text caching

### ISSUE THREAD

**Issue by YehonatanVonage** [typeorm/typeorm#3064]
**Issue type:**
[*] question

**Database system/driver:**
[*] `postgres`

**TypeORM version:**
[*] `0.2.7`

TLDR
the query that I make using TypeORM is cpu intensive and reduce the requests throughput that the server can handle.

I use typeORM in my web server for querying a table with 10 relations.
I do it by configuring an entity with 30 properties that 10 of them includes "eager: true".
In order to query the entities from the DB I use find method.
When I load test my server with requests and profile it, I see in the profiling results that building the query with joins consumes a lot of time (especially the joins), and when I put eager:false on my properties, the request throughput turns to be x2 (doubled)

The queries to the DB are the same except for parameters values which are changed in each request.

Is there a way I can save the object with the query, so it won't build it in each run (except for the parameters change)? 
(I tried also to cache createQueryBuilder with 10 leftJoinAndSelect, with no success).

Thanks!

**Comment by vyrtus15** [comment]
Hello,
We have the same issue using Cordova driver on Android devices, query building with 5-10 joins takes ~400ms, is it possible to cache the query building or boost this part?

**Comment by imnotjames** [comment]
There's a cache for queries now.  If you nee dmore than that you'll need to cache on top of TypeORM.

For other questions, please check out the [community slack](https://join.slack.com/t/typeorm/shared_invite/zt-gej3gc00-hR~L~DqGUJ7qOpGy4SSq3g) or check [TypeORM's documentation page on other support avenues](https://typeorm.io/#/support) - cheers!
